lib/pending_assigns_reminder.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class PendingAssignsReminder
+  def remind(user)
+    topics = assigned_topics(user)
+    return if topics.size < 2
+
+    title = I18n.t('pending_assigns_reminder.title', pending_assignments: topics.size)
+
+    PostCreator.create!(
+      Discourse.system_user,
+      title: title,
+      raw: reminder_body(user, topics),
+      archetype: Archetype.private_message,
+      target_usernames: user.username
+    )
+  end
+
+  private
+
+  def assigned_topics(user)
+    Topic.joins(:_custom_fields).select(:slug, :id, :fancy_title, 'topic_custom_fields.created_at AS assigned_at')
+      .where('topic_custom_fields.name = ? AND topic_custom_fields.value = ?', TopicAssigner::ASSIGNED_TO_ID, user.id.to_s)
+      .order('topic_custom_fields.created_at DESC')
+      .limit(6)
+  end
+
+  def reminder_body(user, topics)
+    newest, oldest = topics.each_slice(3).to_a
+    newest_list = build_list_for(:newest, newest)
+    oldest_list = build_list_for(:oldest, oldest)
+
+    I18n.t(
+      'pending_assigns_reminder.body',
+      pending_assignments: topics.size,
+      assignments_link: "#{Discourse.base_url}/u/#{user.username_lower}/assigned",
+      newest_assignments: newest_list,
+      oldest_assignments: oldest_list,
+      frequency: SiteSetting.remind_assigns
+    )
+  end
+
+  def build_list_for(key, topics)
```

Comment:
Can core provide better facilities here?

Reply:
Couldn't find any helper in core. Am I missing something?

Reply:
Oh what I meant is ... is there any code that can be shuffled in core to make this easier for plugins?

Reply:
I don't believe this will be re-used by another plugin 🤔